### ISSUE TITLE
Wrong Data on INSERT INTO ... SELECT (Please Confirm)

### ISSUE THREAD

**Issue by bhelm** [infinidb/infinidb#13]
i have noticed that infinidb inserts wrong data under some condition when using INSERT INTO .. SELECT syntax without an transaction (so it uses cpimport internally). 
Can someone confirm this Problem? i have tried it with mariadb 4.6.3 and 4.6.4 builds.

Here is how to reproduce:

Create a new database and paste this script:
```
CREATE TABLE `source` (
  `datum` date NOT NULL,
  `datum_hour` datetime NOT NULL,
  `datum_month` date NOT NULL,
  `datum_week` tinyint(3) unsigned NOT NULL,
  `us_id` int(10) unsigned NOT NULL,
  `xy_id` int(10) unsigned NOT NULL,
  `yellow` char(2) DEFAULT NULL,
  `device` tinyint(3) unsigned NOT NULL,
  `device_sys` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `r_id` tinyint(4) NOT NULL DEFAULT '1',
  `os_id` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `br_id` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `op_id` smallint(5) unsigned NOT NULL DEFAULT '0',
  `pa_id` int(10) unsigned NOT NULL,
  `ba_id` int(10) unsigned NOT NULL,
  `rule_id` int(10) unsigned NOT NULL DEFAULT '0',
  `sub_id` varchar(24) DEFAULT NULL,
  `watchme` tinyint(3) unsigned DEFAULT NULL
) ENGINE=InfiniDB DEFAULT CHARSET=latin1 COMMENT='compression=0';

CREATE TABLE `dest` (
  `datum` date NOT NULL,
  `datum_hour` datetime NOT NULL,
  `datum_month` date NOT NULL,
  `datum_week` tinyint(3) unsigned NOT NULL,
  `us_id` int(10) unsigned NOT NULL,
  `xy_id` int(10) unsigned NOT NULL,
  `yellow` char(2) DEFAULT NULL,
  `device` tinyint(3) unsigned NOT NULL,
  `device_sys` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `r_id` tinyint(4) NOT NULL DEFAULT '1',
  `os_id` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `br_id` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `op_id` smallint(5) unsigned NOT NULL DEFAULT '0',
  `pa_id` int(10) unsigned NOT NULL,
  `ba_id` int(10) unsigned NOT NULL,
  `rule_id` int(10) unsigned NOT NULL DEFAULT '0',
  `sub_id` varchar(24) DEFAULT NULL,
  `watchme` tinyint(3) unsigned DEFAULT NULL
) ENGINE=InfiniDB DEFAULT CHARSET=latin1 COMMENT='compression=0';

INSERT INTO source VALUES(
'2015-04-17','2015-04-17 03:00:00','2015-04-01' ,        16,     19,     90,       24          ,     2,         0,      2,    1,   15,  196, 3617,100018,   1114,NULL  ,       9
);

insert into dest select * from source;

select * from source; select * from dest;
```

last select gives that:
```
mysql> select * from source; select * from dest;
+------------+---------------------+-------------+------------+-------+-------+--------+--------+------------+------+-------+-------+-------+-------+--------+---------+--------+---------+
| datum      | datum_hour          | datum_month | datum_week | us_id | xy_id | yellow | device | device_sys | r_id | os_id | br_id | op_id | pa_id | ba_id  | rule_id | sub_id | watchme |
+------------+---------------------+-------------+------------+-------+-------+--------+--------+------------+------+-------+-------+-------+-------+--------+---------+--------+---------+
| 2015-04-17 | 2015-04-17 03:00:00 | 2015-04-01  |         16 |    19 |    90 | 24     |      2 |          0 |    2 |     1 |    15 |   196 |  3617 | 100018 |    1114 | NULL   |       9 |
+------------+---------------------+-------------+------------+-------+-------+--------+--------+------------+------+-------+-------+-------+-------+--------+---------+--------+---------+
1 row in set (0,03 sec)

+------------+---------------------+-------------+------------+-------+-------+--------+--------+------------+------+-------+-------+-------+-------+--------+---------+--------+---------+
| datum      | datum_hour          | datum_month | datum_week | us_id | xy_id | yellow | device | device_sys | r_id | os_id | br_id | op_id | pa_id | ba_id  | rule_id | sub_id | watchme |
+------------+---------------------+-------------+------------+-------+-------+--------+--------+------------+------+-------+-------+-------+-------+--------+---------+--------+---------+
| 2015-04-17 | 2015-04-17 03:00:00 | 2015-04-01  |         16 |    19 |    90 | 24     |      2 |          0 |    2 |     1 |    15 |   196 |  3617 | 100018 |    1114 | NULL   |    NULL |
+------------+---------------------+-------------+------------+-------+-------+--------+--------+------------+------+-------+-------+-------+-------+--------+---------+--------+---------+
1 row in set (0,30 sec)
```
Source and dest do no equal, `watchme` went from 9 to NULL.

I have tried reducing the table, but if i remove more columns, the problem disappears. In my actual setup, my tables have more columns. Within an transaction, this works, but with many rows this is 6 times slower.

Thank You!

**Comment by brianrp** [comment]
If you are using the distribution on Github, InfiniDB installs and uses MySQL 5.1.xx.
You cannot change the database vendor or version independently, or independent of a release of the InfninDB software.
There are changes to the MySQL software to make it work with InfiniDB.

I did test your example and it works correctly with the MySQL version installed with InfiniDB.

If you obtained your version from MariaDB, and they have modified it to use MariaDB in place of MySQL, you will need to contact them for resolution.

Thank you for trying InfinDB.
(Sorry, that’s as far as I can help you, just a user, not a developer.)

**Comment by davidhilldallas** [comment]
Actually to clarify a point with the InfiniDB supported by MariaDB, it still contains MySQL as its front end query engine, not MariaDB.

But there are plans to possible replace it down the road at some point.

**Comment by davidhilldallas** [comment]
I did verify that it is an BUG in the InfiniDB 4.6 builds, it does work in the 4.5 builds, which is also available via the MariaDB InfiniDB Downloads page. A MariaDB InfiniDB BUG has been opened.

Thanks for reporting the issue.

**Comment by davidhilldallas** [comment]
Fix will be in the next MariaDB / InfiniDB release, 4.6.6

Thanks again for reporting issue.

**Comment by fgeller** [comment]
Hi, lurker here: Is there some information available on future releases / plans for InfiniDB / MariaDB?

Thanks!

**Comment by davidhilldallas** [comment]
https://mariadb.com/services/infinidb-services

The support for Enterprise InfiniDB has been picked up by MariaDB since the
Bankrupcy of InfiniDB last year.
They supporting registered customers with MariaDB and are performing BUG
fixes on Issues reported by these Customers. They have been doing monthly
releases, but that is mainly based on what fixes are available.

On Thu, May 7, 2015 at 3:29 PM, Felix Geller <notifications@github.com>
wrote:

> Hi, lurker here: Is there some information available on future releases /
> plans for InfiniDB / MariaDB?
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/infinidb/infinidb/issues/13#issuecomment-100009251>.
>



-- 
------------------
David Hill
InfiniDB Development/Customer Support
MariaDB Corporation
http://www.mariadb.com

**Comment by bhelm** [comment]
Nice, thank you!

**Comment by roninio** [comment]
have the same bug on version SoftwareVersion = 4.6.5.

**Comment by bhelm** [comment]
You can also set infinidb_use_import_for_batchinsert=0 in /usr/local/Calpont/mysql/my.cnf to make sure that the buggy method is not used by accident. dump and cpimport is also what i were doing until i figgured out that the non-cpimport method is faster for my use case.

**Comment by bhelm** [comment]
ok, seems to be resolved with 4.6.6, thanks. :)
